users/test_login_server.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+from django.contrib.auth.models import User
+
+
+USERNAME = "testuser"
+FIRSTNAME = "Test"
+LASTNAME = "User"
+PASSWORD = "testpass"
+EMAIL = "testuser@gmail.com"
+
+
+@pytest.fixture
+def new_user():
+    user = User(username=USERNAME, first_name=FIRSTNAME, last_name=LASTNAME, password=PASSWORD, email=EMAIL)
+    user.set_password(PASSWORD)
+    return user
+
+
+@pytest.mark.django_db()
+class TestProfileServerResponse:
+    # Test that the login webpage loaded
+    def test_login_entry_point(self, client):
+        response = client.get("/login/")
+        assert response.status_code == 200
+
+    # After successfull login the user is redirected to the post page
+    # Check login of admin
+    def test_login_admin(self, client):
+        response = client.post('/login/', {'username': 'admin', 'password': 'admin1234'})
+        assert response.status_code == 302
+        assert response.url == "/"
+
+    # Check login of normal user
+    def test_login_user(self, client, new_user):
+        new_user.save()
+        response = client.post('/login/', {'username': USERNAME, 'password': PASSWORD})
+        assert response.status_code == 302
+        assert response.url == "/"
+
+    # check for redirection from profile page if the user is not logged in
+    def test_redirection_from_profile(self, client, new_user):
+        response = client.get("/profile/")
+        assert response.status_code == 302
+        assert response.url == "/login/?next=/profile/"
+
+    # check for redirection from direct page if the user is not logged in
+    def test_redirection_from_direct(self, client, new_user):
+        response = client.get("/direct/")
+        assert response.status_code == 302
+        assert response.url == "/login/?next=/direct/"
```

Comment:
`new_user` is not used in the test, it can be removed. \

You can test both with a single test with `parametrize` \
Something like:
```suggestion
@pytest.mark.parametrize("url", ["profile", "direct"])
def test_redirection_from_profile(self, client, url):
        response = client.get(f"/{url}/")
        assert response.status_code == 302
        assert response.url == f"/login/?next=/{url}/"
```

Reply:
@abiton1 

What do you thin is more readable:

`assert response.status_code == 302
 assert response.url == f"/login/?next=/{url}/"`

or:

`assert response.status_code == 302 and response.url == f"/login/?next=/{url}/"`

Reply:
Done 😄